org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -11778,29 +11785,16 @@ private void findTypesAndPackages(char[] token, Scope scope, boolean proposeBase
 			}
 		}
 
+		checkCancel();
```

Comment:
I would prefer to not change this, it makes reviewing harder. Maybe as a separate change that improves the code.